File: swedish-cc/2025/kth-challenge/cheaterdetection/cheaterdetection-ints.py
#!/usr/bin/env python3
_, *vs = map(lambda x : int(float(x)*100 + .5), open(0).read().split())
for x in vs:
    cnt = x * 2 // 3

    if (cnt * 3 // 2 == x or (cnt + 1) * 3 // 2 == x):
        print("VALID")
    else:
        print("IMPOSSIBLE")

File: swedish-cc/2025/kth-challenge/cheaterdetection/cheaterdetection-ints.cpp
#include <bits/stdc++.h>
using namespace std;

// NOTE: To get 0.00 on kattis use faster io

int main() {
  cin.tie(0)->sync_with_stdio(0);

  int n;
  cin >> n;
  while (n--) {
    int x;
    cin >> x;
    x *= 100;
    char t; cin >> t;
    int tt; cin >> tt;
    x += tt;

    int cnt = x*2 / 3;

    if (cnt * 3 / 2 == x || (cnt+1) * 3 / 2 == x) {
      cout << "VALID" << '\n';
    } else {
      cout << "IMPOSSIBLE" << '\n';
    }
  }
}
